src/tap/grpc/server.rs

```diff
@@ -125,10 +142,24 @@ where
             }
         };
 
+        let extract = match req.extract.and_then(|ex| ExtractKind::try_from(ex).ok()) {
+            Some(ex) => ex,
+            None => {
+                // If there's no extract field, the request may have been sent
+                // by an older version of the Linkerd control plane. If this is
+                // the case, rather than failing the tap, just do the only
+                // behavior that older control planes know about --- extract
+                // HTTP data without headers.
+                let default = ExtractKind::Http { headers: false };
```

Comment:
consider a `Default` impl for `ExtractKind`? not a blocker

Reply:
I thought about that — when I was writing this, I kind of thought this was clearer because it put the default behavior here in the one place it was actually used, rather than making a user hunt for it? But a default impl might be clearer.